agora/Agora/Governor/Scripts.hs

```diff
@@ -487,48 +493,50 @@ governorValidator as =
                 # phoistAcyclic
                   ( plam
                       ( \((pfield @"value" #) -> value) ->
-                          0 #< psymbolValueOf # patSymbol # value
+                          0 #< psymbolValueOf # atSymbol # value
                       )
                   )
                 # pfromData txInfoF.outputs
 
           pguardC "Output GATs is more than minted GATs" $
             plength # outputsWithGAT #== gatCount
 
+          -- For a given output, check if it contains a single valid GAT
+          -- and whether it correctly belongs to the group.
           let validateGATOutput' :: Term s (PProposalEffectGroup :--> PTxOut :--> PBool)
               validateGATOutput' =
                 phoistAcyclic $
                   plam
                     ( \effects output -> unTermCont $ do
-                        outputF <- pletFieldsC @'["address", "datum", "referenceScript"] output
-
-                        let receiverScriptHash =
+                        outputF <- pletFieldsC @'["address", "datum", "value"] output
+                        PValue value <- pmatchC $ outputF.value
+                        PMap authorityTokens <-
+                          pmatchC $
+                            passertPJust # "validateGATOutput': Must have GAT in GAT output"
+                              #$ plookup # atSymbol # value
+
+                        let tagToken :: Term _ PTokenName
+                            tagToken =
+                              pmaybeData # pconstant "" # plam (pscriptHashToTokenName . pfromData)
+                                #$ psndTuple # effect
+                            receiverScriptHash =
                               passertPJust # "GAT receiver should be a script"
                                 #$ pscriptHashFromAddress # outputF.address
                             effect =
                               passertPJust # "Receiver should be in the effect group"
                                 #$ AssocMap.plookup # receiverScriptHash # effects
-                            hasCorrectReferenceScript =
-                              pmaybeData
-                                # pconstant True
-                                # plam
-                                  ( ( passertPDJust
-                                        # "Output UTXO should have a reference script"
-                                        # outputF.referenceScript
-                                        #==
-                                    )
-                                      . pfromData
-                                  )
-                                # (psndTuple # effect)
+                            valueGATCorrect =
+                              authorityTokens
+                                #== psingleton # (ppairDataBuiltin # pdata tagToken # pdata 1)
```

Comment:
```hs
outputF.value #== psingletonValue # atSymbol # tagToken
```
So you don't have to deconstruct the `PValue` manually.

Reply:
What about ADA? :sweat_smile:

Reply:
Wouldn't we see 0 ada if the utxo has minimum amount of ada?

Reply:
I don't think so? Even still, we probably should allow any amount of ADA to be stored at an effect I think.

Reply:
Hmm, maybe we should write a util function?

Reply:
I mean, it's good for now though. I just can't help wondering if I have made the same assumption somewher else.